rudder-core/src/main/scala/com/normation/rudder/datasources/DataSource.scala

```diff
@@ -39,11 +39,21 @@ package com.normation.rudder.datasources
 
 import org.joda.time.DateTime
 import org.joda.time.Seconds
+import net.liftweb.common._
 
 sealed trait DataSourceType {
   def name : String
 }
 
+object DataSourceType {
+  def apply(name : String) : Box[DataSourceType] = {
+    if (name == ByNodeSourceType.name) {
```

Comment:
I focused on the "By Node" since we want this at least to work. That's a mistake and we should be able to build all DataSourceType with apply